assembly/rlp.ts

```diff
@@ -265,8 +267,8 @@ export function hashBranchNode(branchNodeChildren: Array<usize>): usize {
   // bytes for hashes = len(0xa0 + hash) = 33*branch_num_children
   // bytes for empty nodes (0x80) = (17 - branch_num_children)
 
-  let child_indexes = Array.create<u8>(17)
-  let child_hash_ptrs = Array.create<usize>(17)
+  let child_indexes = new Array<u8>(17)
+  let child_hash_ptrs = new Array<usize>(17)
```

Comment:
I guess this should be 
```suggestion
let child_indexes = new Array<u8>(17)
  let child_hash_ptrs = Array.create<usize>()
  let child_hash_ptrs = new Array<usize>()
```
because later uses `.push` inside if branch

Reply:
Or it should be
```ts
for (let i = 0; i < 17; i++) {
    // read child index
    if (branchNodeChildren[i] > 0) {
      child_indexes[i] = i as u8
      child_hash_ptrs[i] = branchNodeChildren[i]
    }
  }
```
cc @s1na

Reply:
The first variant is correct. But doesn't having the capacity avoid reallocs?

Btw this function has been re-written as `hashBranch`. At some point when it's been replaced wherever it's used we'll remove this `hashBranchNode` function.